File: pl-domain/country-service/src/main/java/com/polylab/countries/application/mapper/CountryDtoMapper.java
package com.polylab.countries.application.mapper;

import com.jxareas.jxcore.mapper.TwoWayMapper;
import com.polylab.countries.application.dto.CountryDto;
import com.polylab.countries.persistence.model.Country;
import lombok.RequiredArgsConstructor;
import org.modelmapper.ModelMapper;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CountryDtoMapper implements TwoWayMapper<CountryDto, Country> {

    private final ModelMapper mapper;


    @Override
    public Country mapTo(CountryDto source) {
        return mapper.map(source, Country.class);
    }

    @Override
    public CountryDto mapFrom(Country destination) {
        return mapper.map(destination, CountryDto.class);
    }
}

File: pl-domain/country-service/src/main/java/com/polylab/countries/app/mapper/CountryDtoMapper.kt
package com.polylab.countries.app.mapper

import com.jxareas.efficax.annotations.Mapper
import com.jxareas.efficax.extensions.mirrorMap
import com.jxareas.efficax.domain.mapper.MirrorMapper
import com.polylab.countries.app.dto.CountryDto
import com.polylab.countries.persistence.model.Country

@Mapper
object CountryDtoMapper : MirrorMapper<CountryDto, Country> by mirrorMap()
